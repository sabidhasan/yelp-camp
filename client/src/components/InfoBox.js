import React from 'react'

const InfoBox = (props) => {
  return (
    <div className='infoBox__info'>
      {props.address ?
        <React.Fragment>
          <span className='infoBox__icon'>📍</span>
          <span className='infoBox__data bold'>{props.address}</span>
          <span className='infoBox__icon'>🚗</span>
          <span className='infoBox__data'><a href="#">Get Directions</a></span>
        </React.Fragment>
      : null}
      {props.phone ?
        <React.Fragment>
          <span className='infoBox__icon'>📞</span>
          <span className='infoBox__data'>{props.phone}</span>
        </React.Fragment>
      : null}
      {props.email ?
        <React.Fragment>
          <span className='infoBox__icon'>✉️</span>
          <span className='infoBox__data'>{props.email}</span>
        </React.Fragment>
      : null}
    </div>
  )
}

export default InfoBox
