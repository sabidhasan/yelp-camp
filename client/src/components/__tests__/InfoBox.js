import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import InfoBox from '../InfoBox'

configure({ adapter: new Adapter() });

describe('<InfoBox />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      address: 'Some Address',
      phone: '123-456-7890',
      email: 'Some Email',
    }
    component = shallow(<InfoBox { ...props } />);
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});
