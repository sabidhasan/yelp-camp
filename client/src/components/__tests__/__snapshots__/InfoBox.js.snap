// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoBox /> LoginForm component snapshot matches what is expected 1`] = `
<div
  className="InfoBox"
>
  <span
    className="InfoBox__icon"
  >
    <span
      aria-label="Address"
      id="InfoBox__address"
      role="img"
    >
      📍
    </span>
  </span>
  <span
    aria-labelledby="InfoBox__address"
    className="InfoBox__data bold"
  >
    Some Address
  </span>
  <span
    className="InfoBox__icon"
  >
    <span
      aria-label="Get directions"
      id="InfoBox__directions"
      role="img"
    >
      🚗
    </span>
  </span>
  <span
    aria-labelledby="InfoBox__directions"
    className="InfoBox__data"
  >
    <a
      href="https://www.google.com/maps/dir/Some Address"
    >
      Get Directions
    </a>
  </span>
  <span
    aria-label="Phone"
    className="InfoBox__icon"
    id="InfoBox__phone"
    role="img"
  >
    📞
  </span>
  <span
    aria-labelledby="InfoBox__phone"
    className="InfoBox__data"
  >
    123-456-7890
  </span>
  <span
    aria-label="Email address"
    className="InfoBox__icon"
    id="InfoBox__email"
    role="img"
  >
    ✉️
  </span>
  <span
    aria-labelledby="InfoBox__email"
    className="InfoBox__data"
  >
    Some Email
  </span>
</div>
`;
